### PR TITLE
meta: add go build_helper

### PR DESCRIPTION
**Pull request by maxice8**

### CODE REVIEW

**Comment by the-maldridge** [comment]
@nilium @bobertlo would really appreciate a second opinion from you guys on this!

**Comment by nilium** [comment]
I can't find anything wrong with this. Looks like a pretty clean move to its helper. Having the make-gopath function has me wondering about what else from the build style can be a helper function that the style just calls, but that's not as important here. I think this looks good.

**Comment by bobertlo** [comment]
This looks fine to me.

**Comment by maxice8** [comment]
should we prefix the functions with the name of the \<build-helper> to avoid conflicts or just assume no function will ever be sourced from 2 different \<build-helpers> with the same name

**Comment by nilium** [comment]
That seems like it's probably worth doing. It's not likely there'll be another `make-gopath` function, but it might help down the road.

**Comment by maxice8** [comment]
On Thu, Feb 28, 2019 at 05:29:12PM +0000, Noel Cower wrote:
> That seems like it's probably worth doing. It's not likely there'll be another `make-gopath` function, but it might help down the road.
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/9229#issuecomment-468362457

@the-maldridge

yay or nay

**Comment by Duncaen** [comment]
If we keep the `hostmakedepends+=" go" ` then please keep it in `build_style`.

Dependencies should be explicit, this helps everyone, that the `build_styles` add them is a mistake.

File: common/build-helper/go.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+export CGO_ENABLED=1
+
+case "$XBPS_TARGET_MACHINE" in
+	*-musl) export GOCACHE="${XBPS_HOSTDIR}/gocache-muslc" ;;
+	*)	export GOCACHE="${XBPS_HOSTDIR}/gocache-glibc" ;;
+esac
+
+hostmakedepends+=" go"
+nostrip=yes
+nopie=yes
```
Comment (Duncaen) [inline common/build-helper/go.sh]: both harmful for templates that build other things than go.

**Comment by Gottox** [comment]
Which packages would benefit from such a change?

**Comment by maxice8** [comment]
@Gottox 

kuberntes-helm won't need to have _convert_arch() anymore, just use $GOARCH
docker will be able to use go:make-gopath

**Comment by maxice8** [comment]
@Gottox 

kuberntes-helm won't need to have _convert_arch() anymore, just use $GOARCH
docker will be able to use go:make-gopath

**Pull request closed without merging**